Implementing movie progress indicators

PaulWessel opened the pull request:
See #2439 for background.  I am implementing those six types with a) as the default.

PaulWessel commented:
I have implemented the six types of movie progress indicators (a-f) and documented them in the movie module (with picture).  Tested reasonably well.  Most have reasonable defaults, so one for instance can add just **-Pa** and get a good result.  The doc script GMT_movie_progress.sh generates the default looks of all 6 types and that file is included in the movie.rst documentation.  As you may know, these features need to be added during the last steps of each frame, hence movie communicates with the main script via environmental variables.

PaulWessel commented:
Here is a movie that shows the six progress indicators in action at the same time:
[Progress indicators](https://www.dropbox.com/s/sns67ab10sr2qqt/GMT_movie_progress.mp4?dl=0)

seisman commented on doc/scripts/GMT_movie_progress.sh:
```diff
@@ -0,0 +1,13 @@
+cat << EOF > map.sh
+gmt begin
+	gmt set MAP_VECTOR_SHAPE 0.75
+	gmt psbasemap -R0/10/0/5 -JX7.6i/3.4i -Bafg -BWSrt+gbeige
```
```suggestion
	gmt basemap -R0/10/0/5 -JX7.6i/3.4i -Bafg -BWSrt+gbeige
```

seisman commented:
The label positions are incorrect for SI units.
![image](https://user-images.githubusercontent.com/3974108/72669818-64952e80-3a04-11ea-93af-e66e381d9658.png)

PaulWessel commented:
Yes, found that last night too.  Need to track down where that is happening.

PaulWessel commented:
Fixed. Note that movie -CHD etc will create slightly different canvas sizes for inch and cm (9.6x5.4" vs 24x13.5 c), but these both rasterize to the same number of pixels.  I updated the PS after running the tests which run in inch mode.

seisman commented:
**+ae** doesn't work well.

![image](https://user-images.githubusercontent.com/3974108/72671074-0de32100-3a13-11ea-964d-327e1a089f67.png)

PaulWessel commented:
Seems OK.  For that particular case we are 0.41 seconds into the movie and we don't display fractional seconds, hence 00.  Use -M30 and you will be 01.  There is the +sscale modifier to scale frame to time (default uses frames per sec) but I need to move it until after the mode is set...

PaulWessel commented:
Thanks.  I improved things further with good default values for everything so **-P**? is all you need to get started.  Detected a few bugs in our movie scripts (4,8), added progress indicators to 3,7, simplified movie **-T** a bit (no need for +w if you imply it via **-L**, **-P**), updated docs for movie with all the defaults, etc.  I have run all the tests and built the docs with updated animations and all look well to me.  I will merge in 45 minutes unless you notice anything odd.
